### PR TITLE
some lint fixes + some reorganization

### PR DESCRIPTION
**Pull request by gagoar**
basic re org like mono repos we use elsewhere. 

tighten some linting rules. I've done the corrections as well. 

I'm still working on more stuff on this tho.

### CODE REVIEW

File: renovate.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (Hermanya) [inline renovate.json:1]: what is renovate.json?

File: LICENSE [hunk 2]
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2020 Herman Starikov
+Copyright (c) 2020 Gago
```
Comment (Hermanya) [inline LICENSE:3]: pretty sure this was unintentional, but https://law.stackexchange.com/questions/16847/how-should-multiple-copyright-holders-be-credited-using-the-mit-license

File: package.json [hunk 3]
```diff
@@ -1,4 +1,113 @@
...
+  },
+  "lint-staged": {
+    "*": [
+      "pretty-quick --staged"
+    ],
+    "*.ts": [
+      "eslint --fix"
+    ]
+  },
+  "prettier": {
```
Comment (Hermanya) [inline package.json:26]: any reason for these prettier changes? I kinda like defaults

File: package.json [hunk 4]
```diff
@@ -1,4 +1,113 @@
...
-  "private": true,	
-  "workspaces": ["collect-package-metadata", "service"]	
-}
\ No newline at end of file
+{
+  "name": "package-analyzer",
+  "private": "true",
+  "repository": "git@github.com:Hermanya/package-analyzer.git",
+  "version": "0.1.8",
+  "description": "Package analyzer 🔎 📦",
```
Comment (Hermanya) [inline package.json:6]: :+1:

File: package.json [hunk 5]
```diff
@@ -1,4 +1,113 @@
...
+    "*.ts": [
+      "eslint --fix"
+    ]
+  },
+  "prettier": {
+    "singleQuote": true,
+    "semi": true,
+    "printWidth": 120
+  },
+  "eslintConfig": {
```
Comment (Hermanya) [inline package.json:31]: you might have noticed that I'm using https://github.com/formium/tsdx which comes with a lot of default configuration. How do these overrides work together?

File: LICENSE (hunk 2 above)
Comment (gagoar) [reply LICENSE:3]: oh, yea, this is cp, from what I had. reverting!

File: package.json (hunk 3 above)
Comment (gagoar) [reply package.json:26]: this is a basic thing we have been using in our all-new projects. no other reason.

File: package.json (hunk 5 above)
Comment (gagoar) [reply package.json:31]: so, In general I remove tsdx given that is too bulky and slow I just this eslint config that is pretty basic, but really strict.

File: renovate.json (hunk 1 above)
Comment (gagoar) [reply renovate.json:1]: this is something I will add as integration so we get packages updated frequently. take a look at my projects. you can see what it does.

Comment (Hermanya) [reply renovate.json:1]: I enabled dependabot for this repo, I think it's a more popular alternative. Please remove this file.

File: package.json (hunk 3 above)
Comment (Hermanya) [reply package.json:26]: This basic thing results in changes that make this particular PR quite noisy.
In addition to this, I do prefer defaults, please remove these changes.

File: package.json (hunk 5 above)
Comment (Hermanya) [reply package.json:31]: I respectfully disagree. 

I'm not sure what you found bulky and slow about it, but I like the "zero-config" nature of this tool. 
Not only it's a huge productivity boost. But it's also very widely adopted and liked within the industry, and endorsed by the typescript team.

It's very much like `create-react-app` (or rather `react-scripts`), in that it abstracts away the configuration of a typical project. Be it react app or a typescript library. The cost of maintenance is dramatically reduced, if you are willing to stay within the "standard configuration", which I intend to.

My plan is to use `tsdx` for the library parts of this project and CRA for the future UI.

**Comment by gagoar** [comment]
I think doing this PR was premature. I will open a new one with fewer changes instead.

**Pull request closed without merging**